soccer/gameplay/skills/benchmark/motion_benchmark.py

```diff
@@ -0,0 +1,390 @@
+                            lambda: self.isSetup, 'Ready to start running')
+
+        #TestMotion -> TestBuffer
+        self.add_transition(MotionBenchmark.State.TestMotion,
+                            MotionBenchmark.State.TestBuffer,
+                            lambda: self.currentTest.motionCompleted() and not self.currentTest.testCompleted(), 'In Position')
+
+        #TestBuffer -> TestMotion
+        #Note that this transition happens instantly, as the buffer is just to reset the state machine, causing on_enter and on_exit to trigger
+        self.add_transition(MotionBenchmark.State.TestBuffer,
+                            MotionBenchmark.State.TestMotion,
+                            lambda: self.currentTest.isMotionPause(), 'In Position')
+
+        #TestMotion -> TestEnd
+        #We have to check both subbehavior completion and testCompleted() because some tests aren't subbehaviors
+        self.add_transition(MotionBenchmark.State.TestMotion,
+                            MotionBenchmark.State.TestEnd, #I should probably change this to just testCompleted(), its a bit of a hack right now
+                            lambda: self.currentTest.testCompleted(), 'In Position')
+
+        #TestEnd -> TestMotion
+        self.add_transition(MotionBenchmark.State.TestEnd,
+                            MotionBenchmark.State.TestMotion,
+                            lambda: self.currentTest != None, 'Next test exists')
+
+
+        #BasicMotionEnd -> ProcessAllTests
+        self.add_transition(MotionBenchmark.State.TestEnd,
+                            MotionBenchmark.State.ProcessAllTests,
+                            lambda: self.testIndex >= len(self.tests) or self.currentTest == None, 'No next test exists')
+                            
+        #ProcessTest -> exit
+        self.add_transition(MotionBenchmark.State.ProcessAllTests,
+                            behavior.Behavior.State.completed,
+                            lambda: True, 'In Position') #Should change this to true if I want it to end
+
+        #Setup the BasicMotionTests
+
+    #Movement test points START
+    setupPoint = robocup.Point(0, 1.25)
+
+
+    #PureRot
+    startPoint = robocup.Point(0,1.5)
+    PureRot0FacePoint = robocup.Point(0, 2.5)
+    PureRot1FacePoint = robocup.Point(-1,1.5)
+    PureRot2FacePoint = robocup.Point(1,1.5)
+
+
+    
+    ##
+    # Appends a test to the list of tests
+    def addTest(self, a):
+        self.tests.append(a)
+
+
+    #A function that both prints and adds to the output file list to be written
+    def resultOut(self, result):
+        print(result)
+        self.resultsToWrite.append(result)
+
+    def fileOnly(self, result):
+        self.resultsToWrite.append(result)
+
+    def checkSub(self):
+        print(self.subbehaviors_by_name)
+
+    ##
+    # This function used to do something
+    #
+    def setupTests(self):
+       pass 
+
+    def getVel(self):
+        return self.robot.vel
+
+    def getSpeed(self):
```

Comment:
lmao i feel like these functions (getSpeed, getAngleError, getPosError) should be implemented in c++ for every robot. correct me if im wrong though